### PR TITLE
Split TrackTPC and its clusters and use span<POD> to receive DPL inputs

### PR DESCRIPTION
**Pull request by shahor02**
@wiechula @davidrohr @matthiasrichter I've eliminated the ``vector<uint32_t>`` from the TrackTPC to make it POD and modified the dependent code.
This allows to use spans instead of vectors for the DPL inputs (hence to avoid copying the input data). 
Together with other vector->span changes this alone saves ``~0.7 GB`` of memory in TPCITS matching component for 50 pbpb events.

@martenole I've also modified your interpolation code to account for modified TrackTPC.

### CODE REVIEW

File: Detectors/TPC/reconstruction/src/GPUCATracking.cxx [hunk 1]
```diff
@@ -68,9 +68,11 @@ int GPUCATracking::runTracking(GPUO2InterfaceIOPtrs* data)
   }
 
   std::vector<TrackTPC>* outputTracks = data->outputTracks;
+  std::vector<uint32_t>* outClusRefs = data->outpuClusRefs;
```
Comment (davidrohr) [inline Detectors/TPC/reconstruction/src/GPUCATracking.cxx]: typo: should be outputClusRefs (also in more places)

**Comment by shahor02** [comment]
@davidrohr any idea why Cuda build fails with 
````
-- 
-- Configuring done
CMake Error in GPU/GPUTracking/Base/cuda/CMakeLists.txt:
  Target "O2lib-GPUTrackingCUDA" requires the language dialect "CUDA17" (with
  compiler extensions), but CMake does not know the compile flags to use to
  enable it.


CMake Error in Detectors/ITSMFT/ITS/tracking/cuda/CMakeLists.txt:
  Target "O2lib-ITStrackingCUDA" requires the language dialect "CUDA17" (with
  compiler extensions), but CMake does not know the compile flags to use to
  enable it.
````

**Comment by davidrohr** [comment]
You pulled in a dependency that requires an interface language of c++17 or higher to one on the libraries that are buil for cuda (can be indirectly over the dependency chain).
I fixed/hacked this manually for root, by removing the interface requirements in root's cmake target, so all root libraries should be ok now.
But there are other libraries in o2 which set these interface properties.
I had a similar problem a month ago with some dependency on fairroot, which i could fix by forward-declaring and removing the dependency in cmake.
Unfortunately, there is now way to fix this before cuda 11 is released, except for getting rid of the dependency, or hacking the exported cmake targets of the offending dependency like i did for root.
Kind Regards
David Rohr

**Comment by shahor02** [comment]
@davidrohr thanks, any hint on how to find which dependencies are problematic?

**Comment by davidrohr** [comment]
Unfortunately not really, i would try 1 by 1 to remove all cmake dependencies you added. (This might break the compilation.)
The good thing is that cmake bails out before it even starts to compile, so you can easily identify the dependency: as soon as it starts compiling something, you removed the problematic one.

Still, that doesn't tell you how to fix it of course.

**Comment by shahor02** [comment]
@davidrohr ok, the offender is ``O2::Framework`` which I added to ``DataFormats/Detectors/ITSMFT/common/CMakeLists.txt`` to have 
https://github.com/AliceO2Group/AliceO2/blob/af5a3d317a261f7ad9aff2c002278a4cca54e88d/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/Cluster.h#L17
and
https://github.com/AliceO2Group/AliceO2/blob/af5a3d317a261f7ad9aff2c002278a4cca54e88d/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/Cluster.h#L191
(related to https://github.com/AliceO2Group/AliceO2/pull/2651 )

Do you see a way to avoid the dependency of ``DataFormats/Detectors/ITSMFT/common/CMakeLists.txt`` from the framework?

**Comment by davidrohr** [comment]
Can't you just forward-declare the template instead of the include:
template <typename T> struct is_messageable;

**Comment by shahor02** [comment]
@MichaelLettrich @ktf the o2 build seems to fail randomly, with the logs of different lengths not showing apparent errors.

**Pull request merged**